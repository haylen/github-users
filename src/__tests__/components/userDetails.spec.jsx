import React from 'react';
import renderer from 'react-test-renderer';

import UserDetails from '../../components/userDetails';

describe('components/userDetails', () => {
  const user = {
    login: 'test login',
    html_url: 'test url',
    avatar_url: 'test avatar',
    created_at: 'test created date',
    public_repos: 12,
    followers: 345,
  };

  it('matches snapshot', () => {
    const tree = renderer.create(
      <UserDetails user={user} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
