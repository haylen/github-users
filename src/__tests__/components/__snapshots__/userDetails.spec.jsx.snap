// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/userDetails matches snapshot 1`] = `
<div
  className="UserDetails"
>
  <img
    alt="test login avatar"
    className="UserDetails__avatar"
    src="test avatar"
  />
  <div
    className="UserDetails__info"
  >
    <a
      className="UserDetails__link"
      href="test url"
      target="_blank"
    >
      Checkout on GitHub
    </a>
    <div
      className="UserDetails__attr"
    >
      Login: 
      test login
    </div>
    <div
      className="UserDetails__attr"
    >
      Number of followers: 
      345
    </div>
    <div
      className="UserDetails__attr"
    >
      Number of public repos: 
      12
    </div>
    <div
      className="UserDetails__attr"
    >
      On GitHub since: 
      Invalid Date
    </div>
  </div>
</div>
`;
